plotly/plotly/plotly.py

```diff
@@ -1342,6 +1345,114 @@ def get_grid(grid_url, raw=False):
     return Grid(parsed_content, fid)
 
 
+class dashboard_ops:
+    """
+    Interface to Plotly's Dashboards API.
+    Plotly Dashboards are JSON blobs. They are made up by a bunch of
+    containers which contain either empty boxes or boxes with file urls.
+    """
+    @classmethod
+    def upload(cls, dashboard, filename, sharing='public', auto_open=True):
+        """
+        BETA function for uploading/overwriting dashboards to Plotly.
+
+        :param (dict) dashboard: the JSON dashboard to be uploaded. Use
+            plotly.dashboard_objs.dashboard_objs to create a Dashboard
+            object.
+        :param (str) filename: the name of the dashboard to be saved in
+            your Plotly account. Will overwrite a dashboard of the same
+            name if it already exists in your files.
+        :param (str) sharing: can be set to either 'public', 'private'
+            or 'secret'. If 'public', your dashboard will be viewable by
+            all other users. If 'secret', only you can see your dashboard.
+            If 'secret', the url will be returned with a sharekey appended
+            to the url. Anyone with the url may view the dashboard.
+        :param (bool) auto_open: automatically opens the dashboard in the
+            browser.
+        """
+        if sharing == 'public':
+            world_readable = True
+        elif sharing == 'private':
+            world_readable = False
+        elif sharing == 'secret':
+            world_readable = False
+
+        data = {
+            'content': json.dumps(dashboard),
+            'filename': filename,
+            'world_readable': world_readable
+        }
+
+        # lookup if pre-existing filename already exists
+        try:
+            v2.files.lookup(filename)
+            matching_dashboard = cls._get_dashboard_json(
+                filename, False
+            )
+
+            if matching_dashboard['filetype'] == 'dashboard':
+                old_fid = matching_dashboard['fid']
+                res = v2.dashboards.update(old_fid, data)
+
```

Comment:
missing an `else` here. 

I'm assuming that `v2.dashboards.update` will fail if the filename already exists as a different type like a `plot`?

Perhaps we can just `throw exceptions.PlotlyRequestError('{filename} is already a {filetype} in your account. While you can overwrite dashboards with the same name, you can't change overwrite files with a different type.\nTry deleting '{filename}' in your account or changing the filename'.format(filename=filename, filetype=matching_dashboard['filetype']')`